src/modal.ts

```diff
@@ -0,0 +1,462 @@
+
+    @property({type:String, attribute:true})
+    height = this.DEFAULT_VALUES.height;
+
+    @property({type:String, attribute:true})
+    width = this.DEFAULT_VALUES.width;
+
+    @property({type:String, attribute:true})
+    animation = this.DEFAULT_VALUES.animation;
+
+    @property({type:String, attribute:true})
+    closesel = null;
+
+    @state()
+    animationDur = .3;
+
+    @state()
+    _position_breakpoints = this.DEFAULT_VALUES.position_breakpoints;
+
+    @state()
+    _height_breakpoints = this.DEFAULT_VALUES.height_breakpoints;
+
+    @state()
+    _width_breakpoints = this.DEFAULT_VALUES.width_breakpoints;
+
+    @state()
+    _animate_breakpoints = this.DEFAULT_VALUES.animate_breakpoints;
+
+    @state()
+    _guide_value:{ [key:string]:any} = {};
+
+    @state()
+    _open = false;
+
+    @state()
+    _mouse_pos_mod = null;
+
+    @state()
+    _change_mc = false;
+
+    @state()
+    timer_obj:ReturnType<typeof setTimeout>|null = null;
+
+    @query('#__wrapper')
+    _wrapper_el:HTMLElement;
+
+    actualParent:HTMLElement|null;
+
+    private _mouse_up_listener:any;
+    private _mouse_move_listener:any;
+    private _mouse_prev_x:number;
+    private _mouse_prev_y:number;
+    private _wrp_init_bounds:any;
+
+    override attributeChangedCallback(name:string, old_val:string,
+                                      new_val:string) {
+        super.attributeChangedCallback(name, old_val, new_val);
+        if (name == 'open') {
+            requestAnimationFrame(() => {
+             this.open?this.show():this.hide()
+            });
+        }
+        if (name == 'height'){
+            this._height_breakpoints = getResponsiveValues(new_val,
+                                       this.DEFAULT_VALUES.height);
+        }
+        if (name == 'width')
+            this._width_breakpoints = getResponsiveValues(new_val,
+                                      this.DEFAULT_VALUES.width);
+        if(name == 'pos') {
+            this._position_breakpoints = getResponsiveValues(new_val,
+                                         this.DEFAULT_VALUES.pos,
+                (v:string) => {
+                    return position_defs[v] != null;
+                });
+        }
+        if(name == 'animation') {
+            this._animate_breakpoints = getResponsiveValues(new_val,
+                                        this.DEFAULT_VALUES.animation,
+                (v:string) => {
+                    return animate_defs[v] != null;
+            });
+        }
+        if(name == 'guidesel') {
+            let tmp = parseAttrVal(new_val);
+            this._guide_value = {}
+            for (let k of Object.keys(tmp))
+                this._guide_value[tmp[k]] = k;
+        }
+    }
+    constructor() {
+        super();
+        document.addEventListener("keydown", this.handleKey.bind(this));
+    }
+    override connectedCallback() {
+        super.connectedCallback();
+    }
+    override disconnectedCallback() {
```

Comment:
We can call `super.disconnectedCallback() ` in here. If not the parent class cleanup might be missed.